bpf/init.sh

```diff
@@ -301,6 +304,38 @@ function bpf_load()
 	return $RETCODE
 }
 
+function bpf_load_cgroups()
+{
+	OPTS=$1
+	IN=$2
+	OUT=$3
+	PROG_TYPE=$4
+	WHERE=$5
+	SEC=$6
+	CALLS_MAP=$7
+	CGRP=$8
+	BPFMNT=$9
+
+	OPTS="${OPTS} ${OPTS_DIR} -DCALLS_MAP=${CALLS_MAP}"
+	bpf_compile $IN $OUT obj "$OPTS"
+
+	TMP=$(mktemp -d $BPFMNT/tmp_XXXXXXXXXX)
+	TMP_FILE="$TMP/x"
+
+	cilium-map-migrate -s $OUT
+	set +e
+	tc exec bpf pin $TMP_FILE obj $OUT type $PROG_TYPE attach_type $WHERE sec $SEC
+	RETCODE=$?
+	set -e
+	cilium-map-migrate -e $OUT -r $RETCODE
+
+	if [ "$RETCODE" -eq "0" ]; then
+		bpftool cgroup attach $CGRP $WHERE pinned $TMP_FILE
+		rm -rf $TMP
```

Comment:
Again if the `cgroup attach` can fail, then this will not be invoked.

Reply:
But in that case it's fine imho, since we can debug it, no?

Reply:
OK, that's probably fine then. Also this shouldn't happen very many times because it's done with the base programs. (More problematic if it was per-endpoint because that could happen a lot)